tests/stages/test_discovery.py

```diff
@@ -25,24 +29,80 @@
 yaml = YAML()
 
 
+@contextlib.contextmanager
+def _caching_for(
+    project: str,
+    stage: Stage = TestStage.build(),
+    hashed_contents: str = "e993ba4f2b2ae2c4840e1eed1414baa812932319d332b0d169365b0885ec2d6c",
+):
+    path = f"tests/projects/{project}/deployment/{RUN_ARTIFACTS_FOLDER}"
+
+    if not os.path.isdir(path):
+        os.makedirs(path)
+
+    try:
+        Output(
+            success=True,
+            message="a test output",
+            produced_artifact=Artifact(
+                artifact_type=ArtifactType.DOCKER_IMAGE,
+                revision="a git revision",
+                producing_step="a step",
+                spec=DockerImageSpec(image="docker-image-path"),
+                hash=hashed_contents,
+            ),
+        ).write(
+            target_path=path,
+            stage=stage.name,
+        )
+        yield path
+    finally:
+        shutil.rmtree(path)
```

Comment:
Can you explain what is happening here? It seems to me you're writing the output to the path and then remove everything in the path again 😅 that would be very weird, so I must be missing something